lib/puppet/parser/functions/str2resource.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+#
+# str2resource.rb
+#
+module Puppet::Parser::Functions
+  newfunction(:str2resource, type: :rvalue, doc: <<-DOC
```

Comment:
Isn't this the legacy API? Any thoughts on using the new Ruby v4 API?

Reply:
I'd be happy to do so, is there some code I can clone from?

Reply:
This module has got [loads of examples](https://github.com/puppetlabs/puppetlabs-stdlib/tree/main/lib/puppet/functions/stdlib).  Only a [few of which](https://github.com/puppetlabs/puppetlabs-stdlib/tree/main/lib/puppet/functions/stdlib) are namespaced, but I think this is preferred for new function.

@binford2k Wrote a [tool for converting functions](https://binford2k.com/2019/11/19/refactoring-legacy-functions/) written in the legacy API.

The API is documented in detail [here](https://github.com/puppetlabs/puppet-specifications/blob/master/language/func-api.md).

What's your use-case for this BTW?

Reply:
I've tried to convert to the new style.  I'd swear my code matches the other working ones, but I'm getting a few errors I don't understand....

As for my use case, I'm working in both Ansible and Puppet these days.  I've grown fairly attached to a simple module that lets me define trivial resources in hiera in a similar way to what I'm doing in Ansible.

psudo code
```puppet
class makeit ( Hash $resources = {} ) {
  $resources.each | $type, $value | {
    create_resources($type, $value)
  }
}
```

When I'm pushing out things like the ssl certificate, it would be handy to be able to set `notify => Class['apache']` and `require => Package[ca-certificates]` so that the service gets information about the state change and things happen in the order I want.  Building an entire module infrastructure for each of my system secrets so they can notify the right classes/services/etc feels like overkill.  I could probably trick the ordering, but this would let me be more explicit in my definitions.

Reply:
@jcpunk Is it as simple as the converted function now not being in the correct directory?

Reply:
:) Indeed, that made a world of difference.